packages/storage/src/database/common/transactions.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  Filter,
+  FindOptions,
+  ReplaceOptions,
+  UpdateResult,
+  Document
+} from 'mongodb';
+import { zkDatabaseConstants } from '../../common/const.js';
+import { DB } from '../../helper/db-instance.js';
+import ModelBasic from '../base/basic.js';
+import ModelCollection from '../general/collection.js';
+
+export type TransactionType = 'deploy' | 'rollup';
+
+export type DbTransaction = {
+  transactionType: TransactionType;
+  databaseName: string;
+  tx: string;
+};
+
+export class ModelDbDeployTx extends ModelBasic<DbTransaction> {
+  private static INSTANCE: ModelDbDeployTx;
+
+  private constructor() {
+    super(
+      zkDatabaseConstants.globalDatabase,
+      DB.service,
+      zkDatabaseConstants.globalCollections.deploy
+    );
+  }
+
+  public static getInstance() {
+    if (!ModelDbDeployTx.INSTANCE) {
+      this.INSTANCE = new ModelDbDeployTx();
+    }
+    return this.INSTANCE;
+  }
+
+  public async create(
+    args: DbTransaction,
+    options?: ReplaceOptions
+  ): Promise<Document | UpdateResult<DbTransaction>> {
+    try {
+      const result = await this.collection.replaceOne(
+        {
+          transactionType: args.transactionType,
+          databaseName: args.databaseName,
+        },
+        args,
+        {...options, upsert: true}
+      );
+
+      return result;
+    } catch (error) {
+      throw new Error(`Failed to create database setting: ${error}`);
+    }
+  }
```

Comment:
no need to using try catch block here
just simply 
```
return this.collection.replaceOne(
        {
          transactionType: args.transactionType,
          databaseName: args.databaseName,
        },
        args,
        {...options, upsert: true}
      );
```

Reply:
done